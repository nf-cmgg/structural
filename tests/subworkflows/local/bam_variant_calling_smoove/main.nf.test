nextflow_workflow {

    name "Test Workflow BAM_VARIANT_CALLING_SMOOVE"
    script "subworkflows/local/bam_variant_calling_smoove/main.nf"
    workflow "BAM_VARIANT_CALLING_SMOOVE"

    tag "subworkflows"
    tag "subworkflows_local"
    tag "bam_variant_calling_smoove"

    test("homo_sapiens") {

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [id:"test", sample:"test", sex:"male"],
                    file(params.cram1, checkIfExists: true),
                    file(params.crai1, checkIfExists: true)
                ])
                input[1] = Channel.value([
                    [id:"fasta"],
                    file(params.fasta, checkIfExists: true)
                ])
                input[2] = Channel.value([
                    [id:"fai"],
                    file(params.fai, checkIfExists: true)
                ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    path(workflow.out.smoove_vcfs[0][1]).linesGzip[-4..-1],
                    file(workflow.out.smoove_vcfs[0][2]).name,
                    workflow.trace.succeeded().size(),
                    workflow.trace.failed().size()
                ).match() }
            )
        }

    }

}
