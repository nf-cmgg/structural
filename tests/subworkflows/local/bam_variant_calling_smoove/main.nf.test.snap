{
    "homo_sapiens": {
        "content": [
            [
                "##FORMAT=<ID=PE,Number=1,Type=Integer,Description=\"Number of paired-end reads supporting the variant\">",
                "##FORMAT=<ID=SR,Number=1,Type=Integer,Description=\"Number of split reads supporting the variant\">",
                "##FORMAT=<ID=BD,Number=1,Type=Integer,Description=\"Amount of BED evidence supporting the variant\">",
                "#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\tFORMAT\tPosCon1"
            ],
            "test.smoove.vcf.gz.tbi",
            6
        ],
        "timestamp": "2023-11-28T14:45:59.166898036"
    }
}